tests/Aggregation/CompositeTest.php

```diff
@@ -0,0 +1,196 @@
+        $composite->addSource((new Terms('color'))->setField('color.keyword'));
+
+        $query = new Query();
+        $query->addAggregation($composite);
+
+        $results = $this->_getIndexForTest()->search($query)->getAggregation('products');
+        $expected = [
+            'after_key' => [
+                'color' => 'red',
+            ],
+            'buckets' => [
+                [
+                    'key' => [
+                        'color' => 'blue',
+                    ],
+                    'doc_count' => 2,
+                ],
+                [
+                    'key' => [
+                        'color' => 'green',
+                    ],
+                    'doc_count' => 1,
+                ],
+                [
+                    'key' => [
+                        'color' => 'red',
+                    ],
+                    'doc_count' => 1,
+                ],
+            ],
+        ];
+
+        $this->assertEquals($expected, $results);
+    }
+
+    /**
+     * @group functional
+     */
+    public function testCompositeWithSizeAggregation(): void
+    {
+        $composite = new Composite('products');
+        $composite->setSize(2);
+        $composite->addSource((new Terms('color'))->setField('color.keyword'));
+
+        $query = new Query();
+        $query->addAggregation($composite);
+
+        $results = $this->_getIndexForTest()->search($query)->getAggregation('products');
+        $expected = [
+            'after_key' => [
+                'color' => 'green',
+            ],
+            'buckets' => [
+                [
+                    'key' => [
+                        'color' => 'blue',
+                    ],
+                    'doc_count' => 2,
+                ],
+                [
+                    'key' => [
+                        'color' => 'green',
+                    ],
+                    'doc_count' => 1,
+                ],
+            ],
+        ];
+
+        $this->assertEquals($expected, $results);
+    }
+
+    /**
+     * @group functional
+     */
+    public function testCompositeWithAfterAggregation(): void
+    {
+        $composite = new Composite('products');
+        $composite->setSize(2);
+        $composite->addSource((new Terms('color'))->setField('color.keyword'));
+        $composite->addAfter(['color' => 'green']);
+        $query = new Query();
+        $query->addAggregation($composite);
+
+        $results = $this->_getIndexForTest()->search($query)->getAggregation('products');
+        $expected = [
+            'after_key' => [
+                'color' => 'red',
+            ],
+            'buckets' => [
+                [
+                    'key' => [
+                        'color' => 'red',
+                    ],
+                    'doc_count' => 1,
+                ],
+            ],
+        ];
+
+        $this->assertEquals($expected, $results);
+    }
+
+    protected function _getIndexForTest(): Index
```

Comment:
This can probably be private instead, and `_` prefix seems superfluous but it's already used in the project, ping @ruflin for confirmation

Reply:
indeed, but it is so in the other tests

Reply:
Yeah, good question. I think we should move away from it but it is a hard one to do.

My suggestion as this is only a test method: Lets not use it.